datafusion/datasource-parquet/src/opener.rs

```diff
@@ -524,6 +511,91 @@ fn should_enable_page_index(
             .unwrap_or(false)
 }
 
+/// Prune based on partition values and file-level statistics.
+pub struct FilePruner {
+    predicate: Arc<dyn PhysicalExpr>,
+    pruning_schema: Arc<Schema>,
```

Comment:
Could we maybe add some comments about what a `pruning_schema` is? And how it relates to `partition_fields`